Common/Securities/SecurityCache.cs

```diff
@@ -97,6 +92,29 @@ public class SecurityCache
         /// </summary>
         public long OpenInterest { get; private set; }
 
+        /// <summary>
+        /// Add a list of market data points to the local security cache for the current market price.
+        /// </summary>
+        /// <remarks>Internally uses <see cref="AddData"/> using the last data point of the provided list
+        /// and it stores by type the non fill forward points using <see cref="StoreData"/></remarks>
+        public void AddDataList(IReadOnlyList<BaseData> data, Type dataType, bool? containsFillForwardData = null)
+        {
+            var nonFillForwardData = data;
+            // maintaining regression requires us to NOT cache FF data
+            if (!containsFillForwardData.HasValue || containsFillForwardData.Value)
+            {
+                nonFillForwardData = data.Where(baseData => !baseData.IsFillForward).ToList();
```

Comment:
I'd be curious how often we come down here -- if it's really often, might want to break this out into a `for` and manually create the list using the total size to seed it. This will prevent a potential extra list allocation when there's more than four items in the list (four is the default list capacity IIRC) and there's a surprising amount of overhead on LINQ methods, enumerators, function delegate invocation, stack popping up and down, etc...

Reply:
Good idea - will do 👍